### PR TITLE
Feature #30 create update for countries

### PR DESCRIPTION
**Pull request by aytiqaqash**

### CODE REVIEW

**Review by hamidsultanzadeh — APPROVED** [review]

File: src/main/java/com/webperside/courseerpbackend/controller/CountryController.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/v1/countries")
+public class CountryController {
+
+    private final CountryServiceImpl countryService;
+
+    @PostMapping
+    public void insert(@RequestBody Country country){
```
Comment (hamidsultanzadeh) [inline src/main/java/com/webperside/courseerpbackend/controller/CountryController.java:16]: don't use Mybatis model for accepting payload
create new class for that, f.e CountryPayload

**Review by hamidsultanzadeh — CHANGES_REQUESTED** [review]

File: src/main/java/com/webperside/courseerpbackend/models/mappers/CountryEntityMapper.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import org.mapstruct.MappingTarget;
+import org.mapstruct.factory.Mappers;
+
+@Mapper
+public interface CountryEntityMapper {
+    CountryEntityMapper INSTANCE = Mappers.getMapper(CountryEntityMapper.class);
+
+
+    Country toEntity(CountryPayload countryPayload);
+    Country toEntity(CountryPayload countryPayload, @MappingTarget Country country);
```
Comment (hamidsultanzadeh) [inline src/main/java/com/webperside/courseerpbackend/models/mappers/CountryEntityMapper.java:16]: remove that method. we'll map our payload model to mybatis model and send it to service directly

File: src/main/java/com/webperside/courseerpbackend/services/country/CountryBusinessService.java
Comment (hamidsultanzadeh) [inline src/main/java/com/webperside/courseerpbackend/services/country/CountryBusinessService.java:1]: I know I wrote the description that we'll use business service for that. but after implementing it, I guess we don't need business service. so you can follow this way 

Controller -> Use mappers to map payload to mybatis model -> countryService
don't use business service

**Review by hamidsultanzadeh — CHANGES_REQUESTED** [review]

File: src/main/java/com/webperside/courseerpbackend/models/mappers/CountryEntityMapper.java (hunk 2 above)
Comment (aytiqaqash) [reply src/main/java/com/webperside/courseerpbackend/models/mappers/CountryEntityMapper.java:16]: Do you mean to remove toEntity? So how would I use countryPayload without mapping?

**Review by hamidsultanzadeh — APPROVED** [review]

**Pull request merged**